awx/main/credential_plugins/hashivault.py

```diff
@@ -98,27 +104,30 @@ def kv_backend(**kwargs):
     sess = requests.Session()
     sess.headers['Authorization'] = 'Bearer {}'.format(token)
 
+    request_url = '/'.join([url, secret_path]).rstrip('/')
     if api_version == 'v2':
         if kwargs.get('secret_version'):
             request_kwargs['params'] = {'version': kwargs['secret_version']}
-        try:
-            mount_point, *path = pathlib.Path(secret_path.lstrip(os.sep)).parts
-            '/'.join(path) 
-        except Exception:
-            mount_point, path = secret_path, []
-        # https://www.vaultproject.io/api/secret/kv/kv-v2.html#read-secret-version
-        request_url = '/'.join([url, mount_point, 'data'] + path).rstrip('/')
-        response = sess.get(request_url, **request_kwargs)
-
-        response.raise_for_status()
-        json = response.json()['data']
-    else:
-        request_url = '/'.join([url, secret_path]).rstrip('/')
-        response = sess.get(request_url, **request_kwargs)
-
-        response.raise_for_status()
-        json = response.json()
+        if 'data' not in secret_path.split('/'):
```

Comment:
Alternative approaches would be:

- add a new input field for the mount point and require it for everything
- or, fail loudly in the plugin when no "data" path is found in the secret path for v2 lookups (instead of handling it as a special case like we're doing here)

I'm open to these but they would break existing secret lookups and require users to update their credentials.